src/core/client.ts

```diff
@@ -303,9 +303,30 @@ export class OakConsentClient implements ConsentClient {
     this.hasLoggedFirstVisit = true;
 
     try {
-      await this.networkClient.logUser(this.userId, this.appSlug);
+      await this.networkClient.logUser(
+        this.userId,
+        this.appSlug,
+        this.extractLocationParams(),
+      );
     } catch (error) {
       this.onError(error);
     }
   };
+
+  /**
+   * Extracts additional context from the current page
+   * to be logged with the user's visit
+   */
+  private extractLocationParams() {
+    const url = new URL(window.location.href);
+    return {
+      utmSource: url.searchParams.get("utm_source") ?? undefined,
```

Comment:
Edge case but if utm_source etc is present but without value, this will be set as empty string rather than undefined here. I think that's desired, but just noted that behaviour differs for referrerUrl

Reply:
Good catch! I think it is entirely likely that a bad tool/copy-paste job will leave empty values behind